mixer/pkg/api/envoyGrpcServer.go

```diff
@@ -0,0 +1,258 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"google.golang.org/grpc/codes"
+	grpc "google.golang.org/grpc/status"
+
+	accesslog "github.com/envoyproxy/go-control-plane/envoy/service/accesslog/v2"
+	authz "github.com/envoyproxy/go-control-plane/envoy/service/auth/v2"
+	rpcstatus "google.golang.org/genproto/googleapis/rpc/status"
+
+	rpc "istio.io/gogo-genproto/googleapis/google/rpc"
+	"istio.io/istio/mixer/pkg/attribute"
+	"istio.io/istio/mixer/pkg/checkcache"
+	"istio.io/istio/mixer/pkg/loadshedding"
+	"istio.io/istio/mixer/pkg/runtime/dispatcher"
+	"istio.io/istio/mixer/pkg/status"
+	attr "istio.io/pkg/attribute"
+	"istio.io/pkg/pool"
+)
+
+type (
+	GrpcServerEnvoy struct {
+		dispatcher dispatcher.Dispatcher
+		gp         *pool.GoroutinePool
+		cache      *checkcache.Cache
+
+		// the global dictionary. This will eventually be writable via config
+		globalWordList []string
+		globalDict     map[string]int32
+
+		// load shedding
+		throttler *loadshedding.Throttler
+	}
+)
+
+func NewGRPCServerEnvoy(dispatcher dispatcher.Dispatcher, gp *pool.GoroutinePool, cache *checkcache.Cache,
```

Comment:
nit: add a comment on what this method is doing and why it exists.

Reply:
Done